lib/proxy/settings/global.rb

```diff
@@ -53,5 +53,51 @@ def apply_argv(args)
       self.daemon = true if args.include?('--daemonize')
       self.daemon = false if args.include?('--no-daemonize')
     end
+
+    def apply_env(env)
+      DEFAULT_SETTINGS.each do |setting_key, setting_value|
+        env_key = "PROXY_#{setting_key.to_s.upcase}"
```

Comment:
It may be longer, but maybe `FOREMAN_PROXY_` or `SMART_PROXY_`? Proxy can be ambiguous and I wonder if it can interfere with something.